hatch_vcs/version_source.py

```diff
@@ -53,7 +53,9 @@ def construct_setuptools_scm_config(self):
         config = deepcopy(self.config_raw_options)
         config.setdefault('root', self.root)
 
-        config.setdefault('tag_regex', self.config_tag_pattern)
+        # Only set for non-empty strings
+        if self.config_tag_pattern:
```

Comment:
i believe instead of setdefault the normal setitem operation should be used

Reply:
80a819154c8acbddde6bdc9be15d92ac69dc1d6d I also added a test that goes into the `if` block